mlir/tools/mlir-pet/Lib/mlirCodegen.cc

```diff
@@ -129,26 +129,33 @@ MLIRCodegen::getSymbolInductionVar(__isl_keep pet_expr *expr,
 // TODO: check that expr is freed for all the possible
 // exit points. Do we want also to free in case we return
 // nullptr? Can we do a C++ wrapper around pet_expr?
-// FIXME: pet allows also expression that are like:
-// ref_id: __pet_ref_3
-// index: { S_2[] -> [(123)] }
-// depth: 1
-// read: 1
-// write: 0
-// to be of type pet_expr_access. This will
-// trigger 'tuple has not id' in isMultiDimensionalArray.
-// To reproduce use:
-// int main() {
-//  int i = 10;
-//  i = i + 23;
-// }
 
 Value MLIRCodegen::createLoad(__isl_take pet_expr *expr) {
   LLVM_DEBUG(dbgs() << __func__ << "\n");
   assert((pet_expr_get_type(expr) == pet_expr_access) &&
          "expect pet_expr_access type");
 
   auto location = builder_.getUnknownLoc();
+  //is const ->  no id? //FIXME: handle types other than int
+  auto id = pet_expr_access_get_id(expr);
```

Comment:
you can use: auto id = isl::manage(pet_expr_access_get_id(expr)). So you don't need to free at line 158.